oracle/src/default_combine_data.rs

```diff
@@ -0,0 +1,41 @@
+use rstd::prelude::Vec;
+use support::{parameter_types, traits::Time};
+use traits::CombineData;
+
+use crate::{MomentOf, TimestampedValue, Trait};
+
+parameter_types! {
+	const MinimumCount: usize = 3;
+}
+
+pub struct DefaultCombineData<T: Trait>(rstd::marker::PhantomData<T>);
+
+impl<T: Trait> CombineData<T::Key, TimestampedValue<T::Value, MomentOf<T>>> for DefaultCombineData<T> {
+	fn combine_data(
+		_key: &T::Key,
+		values: Vec<TimestampedValue<T::Value, MomentOf<T>>>,
+		prev_value: Option<TimestampedValue<T::Value, MomentOf<T>>>,
+	) -> Option<TimestampedValue<T::Value, MomentOf<T>>> {
+		let mut valid_values = values
+			.into_iter()
+			.filter_map(|x| {
+				// TODO: check expiry
```

Comment:
still need to do this

Reply:
@xlc you mean I need to implement it on this PR?

Reply:
yes